### DISCUSSION TITLE
Drawing outside of the window?

### DISCUSSION THREAD

**Discussion by seclerp** [raysan5/raylib#3148] [Q&A] [answered]
Hi!

I'm using ImGui with Raylib as my graphical backend. ImGui support a feature called ["Multi-viewports"](https://github.com/ocornut/imgui/wiki/Multi-Viewports), that allows to render data outside of main window's bounds, e.g.:

![A screenshot](https://user-images.githubusercontent.com/8225057/97542423-fe8ffb80-19c6-11eb-9bf5-e26d86364e55.png)

But I can't find any information on achieving this behavior using Rlgl and Raylib windowing.

P.S. I'm a bit newbie in advanced windowing/graphics so my question may sound obvious 😅

**Comment by orcmid** [comment, marked as answer]
This is not achievable directly with raylib windowing.  Ramon has a description of modifications that might allow this to be done.  You would then be maintaining a custom version of raylib.  Out of the box, raylib has only one application window managed with a global static structure shared among all the raylib API operations.  The other way is to draw mini-windows within that bigger window.  No pop-outs or detached dialogs.  Seems like much work.  

I could not find anywhere in the IMGUI projects I have examined where raylib is usable as a back end.  Where do you see that?  I do see that glfw can be a back end to an IMGUI though.  I also suspect that IMGUI being C++ (not compatible Clean C) code raises further challenges.  

Have you built that example?  Perhaps it gives you a preferable choice for the kind of GUI applications you want to make?

**Reply by seclerp** [reply]
I very like Raylib for simplicity of drawing primitives and textures to a render target, but seems like it's not suitable for such advanced scenarios, such as custom windowing, input, ...

Unfortunately Raylib patching is not an option for me, because in such case I also need to maintain a patched bindings, because I'm using .NET instead of C/C++.

Sp after some investigation it seems like it's better to use SDL2, it's the closest competitor to Raylib in terms of rendering simplicity + it provides a way more customization options + it's officially supported by ImGui as a backend.

Anyway, thanks for a detailed answer!

**Comment by raysan5** [comment]
@seclerp By design raylib does not support multiple windows but here there are some details about what is required to implement them: https://github.com/raysan5/raylib/wiki/Use-multiple-windows

**Reply by seclerp** [reply]
Thanks for that, but due to fact that I'm using binding I need to patch them too, so Raylib is not an option at all 😔
